test/unit/gateways/sum_up_test.rb

```diff
@@ -0,0 +1,361 @@
+require 'test_helper'
+
+class SumUpTest < Test::Unit::TestCase
+  def setup
+    @gateway = SumUpGateway.new(
+      token: 'sup_sk_ABC123',
+      pay_to_email: 'example@example.com'
+    )
+    @credit_card = credit_card
+    @amount = 100
+
+    @options = {
+      payment_type: 'card',
+      billing_address: address,
+      description: 'Store Purchase'
+    }
+  end
+
+  def test_successful_purchase
+    @gateway.expects(:ssl_request).returns(successful_create_checkout_response)
+    @gateway.expects(:ssl_request).returns(successful_complete_checkout_response)
+    response = @gateway.purchase(@amount, @credit_card, @options)
+
+    assert_success response
+
+    assert_equal 'PENDING', response.message
+    assert response.test?
+  end
+
+  def test_failed_purchase
+    @credit_card.year = 2020
```

Comment:
💬 Comment:

GIven that `failed_complete_checkout_response` is fixed changing the year is not going to have any effect